astronomer/providers/amazon/aws/example_dags/example_sagemaker.py

```diff
@@ -203,20 +214,82 @@ def set_up(role_arn: str) -> None:
     ti.xcom_push(key="transform_job_name", value=transform_job_name)
 
 
-@task(trigger_rule=TriggerRule.ALL_DONE)
-def delete_logs() -> None:
+def delete_logs(task_instance: "TaskInstance") -> None:
     """Delete the cloud watch log based on the log group name"""
+    import boto3
+    from botocore.exceptions import ClientError
+
     generated_logs = [
         "/aws/sagemaker/ProcessingJobs",
         "/aws/sagemaker/TrainingJobs",
         "/aws/sagemaker/TransformJobs",
     ]
+    sagemaker_credentails_xcom = task_instance.xcom_pull(
+        key="sagemaker_credentials", task_ids=["get_aws_sagemaker_session_details"]
+    )[0]
+    creds = {
+        "aws_access_key_id": sagemaker_credentails_xcom["AccessKeyId"],
+        "aws_secret_access_key": sagemaker_credentails_xcom["SecretAccessKey"],
+        "aws_session_token": sagemaker_credentails_xcom["SessionToken"],
+        "region_name": AWS_DEFAULT_REGION,
+    }
+    client = boto3.client("logs", **creds)
+    for group in generated_logs:
+        try:
+            if client.describe_log_streams(logGroupName=group)["logStreams"]:
+                client.delete_log_group(logGroupName=group)
+        except ClientError as e:
+            raise e
+
+
+def get_aws_sagemaker_session(task_instance: "TaskInstance") -> None:
+    """Get session details by using env variables credentials details"""
     import boto3
+    from botocore.exceptions import ClientError
 
-    client = boto3.client("logs")
-    for group in generated_logs:
-        if client.describe_log_streams(logGroupName=group)["logStreams"]:
-            client.delete_log_group(logGroupName=group)
+    client = boto3.client("sts", **AWS_SAGEMAKER_CREDS)
+    try:
+        response = client.get_session_token(DurationSeconds=10800)
```

Comment:
10800 seconds is risky.

```suggestion
        response = client.get_session_token(DurationSeconds=1200)
```

Reply:
In my local, the Dag completion took 22 min, so just keeping it 30 min